mysql-test/suite/innodb/t/temp_table_encrypt.test

```diff
@@ -0,0 +1,93 @@
+# test temporary tables encryption
+
+--source include/have_innodb.inc
+
+CREATE TEMPORARY TABLE t01 (a INT) ENGINE=InnoDB;
+INSERT INTO t01 VALUES (1), (2), (3);
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t02 (a INT) ENGINE=InnoDB ENCRYPTION='Y';
+
+CREATE TEMPORARY TABLE t03 (a INT) ENGINE=InnoDB ENCRYPTION='N';
+INSERT INTO t03 VALUES (1), (2), (3);
+
+# this table created in separate file per table tablespace
+CREATE TEMPORARY TABLE t04 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED;
+INSERT INTO t04 VALUES (1), (2), (3);
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t05 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='y';
+
+CREATE TEMPORARY TABLE t06 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='n';
+INSERT INTO t06 VALUES (1), (2), (3);
+
+# test that we can turn encryption OFF and ON
+
+SET GLOBAL innodb_temp_tablespace_encrypt = OFF;
+
+CREATE TEMPORARY TABLE t07 (a INT) ENGINE=InnoDB;
+INSERT INTO t07 VALUES (1), (2), (3);
+
+SET GLOBAL innodb_temp_tablespace_encrypt = ON;
+
+# make sure encryption cannot be be turned ON/OFF for session
```

Comment:
"be be", although I'd just leave this part for a sys_vars test and remove it from here altogether

Reply:
removed